packages/jsts/tests/rules/index.test.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * SonarQube JavaScript Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { rules as mapping } from '../../src/rules';
+import fs from 'fs';
+import path from 'path';
+
+describe('index', () => {
+  it('should map keys to rules definitions', () => {
+    // FIXME: This test runs with a time complexity of O(n^2) where n is the number of rules.
+    //        This is because it iterates over all rules and checks if they are mapped. Once
+    //        we use Sonar rule keys as ESLint rule keys, we can make this test O(n) just by
+    //        checking if the rule key is in the mapping.
+    const ruleFolder = path.join(__dirname, '../../src/rules');
+    const sonarKeys = fs.readdirSync(ruleFolder).filter(name => /^S\d+/.test(name));
+    const mappedRules = Object.values(mapping);
+    const missing = [];
+    for (const sonarKey of sonarKeys) {
+      const { rule } = require(path.join(ruleFolder, sonarKey));
+      if (!mappedRules.some(mapped => mapped === rule)) {
+        missing.push(sonarKey);
+      }
```

Comment:
why wouldn't this be enough?

```suggestion
      if (!mappedRules[sonarKey]) {
        missing.push(sonarKey);
      }
```

Reply:
The check we want to implement is that we don't forget to add the rule redefinition/decoration/whatever, so it's the case where we:
1. write it, and its units tests in `packages/src/ rules/Sxxxx/`
2. forget to load it in `packages/src/rules/index.ts`

No?

Reply:
As I was saying in my comment, we are not able to do what you are suggesting because the keys in the mapping aren't Sonar keys:

https://github.com/SonarSource/SonarJS/blob/0b48c18aaad06886ecb1733d4ed55e5fc5d57a77/packages/jsts/src/rules/index.ts#L300-L302

We don't have `rules['S2376'] = S2376; ` but `rules['accessor-pairs'] = S2376;` instead. This is why we need to check the other way around.

Reply:
thanks for the explanation

Reply:
You're welcome :)